File: python/baiduads-sdk/baiduads_manual/oauth/model/user_info.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Copyright (c) 2022 Baidu.com, Inc. All Rights Reserved
"""

from baiduads_manual.oauth.model.sub_user_info import SubUserInfo


class UserInfo:
    """user info data model

    Attributes:
        masterUid: 同意授权用户ucid
        masterName: 同意授权用户对应的ucname
        subUserList: 同意授权用户关联的子账号列表
    """
    def __init__(self, masterUid, masterName, subUserList):
        self.masterUid = masterUid
        self.masterName = masterName
        self.subUserList = subUserList

    @classmethod
    def dict_to_obj(cls, dict_data):
        """用于将dict转换为对象
        """
        sub_user_list = None
        if "subUserList" in dict_data:
            sub_user_list = []
            for d in dict_data["subUserList"]:
                sub_user_list.append(SubUserInfo.dict_to_obj(d))

        return UserInfo(
            dict_data.get("masterUid", None),
            dict_data.get("masterName", None),
            sub_user_list
        )

File: java/baiduads-sdk/src/main/java/com/baidu/dev2/api/sdk/manual/oauth/model/UserInfo.java
package com.baidu.dev2.api.sdk.manual.oauth.model;


import java.util.List;

public class UserInfo {
    /**
     * 同意授权用户ucid
     */
    private Integer masterUid;
    /**
     * 同意授权用户对应的ucname
     */
    private String masterName;
    /**
     * 同意授权用户关联的子账号列表
     */
    private List<SubUserInfo> subUserList;

    public Integer getMasterUid() {
        return masterUid;
    }

    public void setMasterUid(Integer masterUid) {
        this.masterUid = masterUid;
    }

    public String getMasterName() {
        return masterName;
    }

    public void setMasterName(String masterName) {
        this.masterName = masterName;
    }

    public List<SubUserInfo> getSubUserList() {
        return subUserList;
    }

    public void setSubUserList(List<SubUserInfo> subUserList) {
        this.subUserList = subUserList;
    }
}
